dashboard/app/helpers/users_helper.rb

```diff
@@ -21,10 +21,22 @@ def check_and_apply_oauth_takeover(user)
       end
 
       existing_account.destroy! if existing_account
-      user.provider = session['clever_link_flag']
-      user.uid = uid
-      user.oauth_token = session['clever_takeover_token']
-      user.save
+      if user.migrated?
+        user.add_credential(
+          type: session['clever_link_flag'],
+          id: uid,
+          email: user.email,
+          hashed_email: user.hashed_email,
+          data: {
+            oauth_token: session['clever_takeover_token']
+          }
+        )
```

Comment:
We're not doing anything with the return value of `add_credential` which should indicate whether the db interaction succeeded.  Should we make sure we raise or recover somehow if it fails?